From a91072605351d166186368f70336d3c1186e0d0d Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@intel.com>
Date: Fri, 8 May 2015 17:48:51 +0300
Subject: [PATCH 09/11] iio: configfs: Small cleanups

Remove checking in trigger_drop group since it always be non-NULL
and use WARN_ONCE instead of pr_err to avoid spamming the log.

This changes are made after review on iio-list.

Change-Id: If4b696233a8815fb15ded130331d017a30f46a20
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26476
Signed-off-by: Daniel Baluta <daniel.baluta@intel.com>
Reviewed-on: https://android.intel.com:443/486302
---
 drivers/iio/industrialio-configfs.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/iio/industrialio-configfs.c b/drivers/iio/industrialio-configfs.c
index 0361434..daf318c 100644
--- a/drivers/iio/industrialio-configfs.c
+++ b/drivers/iio/industrialio-configfs.c
@@ -33,7 +33,8 @@ static struct config_group *trigger_make_group(struct config_group *group,
 	type_name = buf;
 	trigger_name = strchr(buf, '-');
 	if (!trigger_name) {
-		pr_err("Unable to locate '-' in %s. Use <type>-<name>.\n", buf);
+		WARN_ONCE(1, "Unable to locate '-' in %s. Use <type>-<name>.\n",
+			  buf);
 		return ERR_PTR(-EINVAL);
 	}
 
@@ -55,8 +56,7 @@ static void trigger_drop_group(struct config_group *group,
 {
 	struct iio_sw_trigger *t = to_iio_sw_trigger(item);
 
-	if (t)
-		iio_sw_trigger_destroy(t);
+	iio_sw_trigger_destroy(t);
 	config_item_put(item);
 }
 
-- 
1.9.1

